### PR TITLE
FIX: put current page at top of ClassName dropdown (was broken)

### PR DESCRIPTION
**Pull request by sunnysideup**
## Description
For the "Page Type" field in the settings tab (main) of a page, the current Page Type is supposed to be on top (see code changed). The code did not actually do that though and instead just placed the "last" entry in the list on top. That does not make any sense so I fixed it. 

## Manual testing steps
Open any SS website, go to the settings of a page and look at the list of Page Type options. You will see they are in alphabetical order, but the last one (e.g. the one that starts with last letter in the alphabet is listed first). The current page type is supposed to be listed first!

## Issues
https://github.com/silverstripe/silverstripe-cms/issues/3005

Also see:
https://github.com/silverstripe/silverstripe-cms/issues/2209
https://github.com/silverstripe/silverstripe-cms/issues/2208

## Pull request checklist

- [x] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [x] This change is covered with tests (or tests aren't necessary for this change)
- [x] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

### CODE REVIEW

**Comment by GuySartorelli** [comment]
The "Issues" section is for you to link to an issue. This is made clear in the pull request template:

> List all issues here that this pull request fixes/resolves.
> If there is no issue already, create a new one! You must link your pull request to at least one issue.

Please create a GitHub issue if there isn't one already and link to it. If there's already an issue, please link to that.

**Comment by GuySartorelli** [comment]
You've also put "[ N]" next to "This change is covered with tests (or tests aren't necessary for this change)"

Does this mean you think tests are needed, but haven't included them?
Please either include tests, or explain why tests aren't needed.

**Comment by sunnysideup** [comment]
```
Does this mean you think tests are needed, but haven't included them?
Please either include tests, or explain why tests aren't needed.
```

Test are nice, but I dont have time for it and I dont think such a simple piece of code needs a test (famous last words), partially because it has not had a test so far, AFAIK.

**Comment by sunnysideup** [comment]
In Terms of a test, I think it would be more valuable for a human to see that the UX feels good and works nicely.  I was thinking that you could have:
```
-- MyPageType (current type) -- 
OtherPageType1
OtherPageType2
```
etc...
So that you can always see what the current one is.

**Comment by GuySartorelli** [comment]
The point of unit tests is primarily to prevent regressions. This is a perfect example of something that was working at some point, but due to lack of unit tests it broke at some stage without anyone noticing. 

To prevent this breaking again, it would be best to have a unit test.

**Comment by sunnysideup** [comment]
What exactly would you test for? Would you test that it returns an array or that it returns a valid array?  @GuySartorelli  - I'd be curious if you had any ideas on what exactly to test for.

**Comment by sunnysideup** [comment]
I have added a bunch of tests.

File: tests/php/Model/SiteTreeTest.php [hunk 1]
```diff
@@ -1380,10 +1384,23 @@ public function testClassDropdown()
...
+            $otherPage = new $className();
+            $otherPage->write();
+            $result = $method->invoke(object: $otherPage);
+            $this->assertEquals(array_key_first($result), $className);
+            // remove the first element as this is not alphabetical
+            array_shift($result);
+            // create a sorted array
+            $resultSorted = $result;
+            asort($resultSorted);
+            $this->assertEquals(implode('_', $result), implode('_', $resultSorted));
```
Comment (michalkleiner) [inline tests/php/Model/SiteTreeTest.php]: You don't need to stringify the arrays, assertEquals would compare them, including the order of elements.

File: tests/php/Model/SiteTreeTest.php [hunk 2]
```diff
@@ -1359,6 +1359,10 @@ public function testAllowedChildrenValidation()
         );
     }
 
+    /**
+     * run like this: vendor/bin/phpunit --filter testClassDropdown vendor/silverstripe/cms/tests/php/Model/SiteTreeTest.php
+     * @return void
+     */
```
Comment (michalkleiner) [inline tests/php/Model/SiteTreeTest.php]: Probably not needed

File: tests/php/Model/SiteTreeTest.php [hunk 3]
```diff
@@ -1380,10 +1384,23 @@ public function testClassDropdown()
...
+            // create a sorted array
+            $resultSorted = $result;
+            asort($resultSorted);
+            $this->assertEquals(implode('_', $result), implode('_', $resultSorted));
 
+        }
         Security::setCurrentUser(null);
     }
 
+
```
Comment (michalkleiner) [inline tests/php/Model/SiteTreeTest.php]: unrelated change

**Comment by michalkleiner** [comment]
Just passing by, left a few minor comments. Thanks for taking the time to add the tests as well, @sunnysideup!

File: tests/php/Model/SiteTreeTest.php [hunk 4]
```diff
@@ -1359,6 +1359,9 @@ public function testAllowedChildrenValidation()
         );
     }
 
+    /**
+     * @return void
+     */
```
Comment (ChloeSartorelli) [inline tests/php/Model/SiteTreeTest.php:1364]:
```suggestion
```
We don't need this typehint, but I won't delay merging just for this.

**Review by GuySartorelli — APPROVED** [review]
LGTM and works well locally.
Thank you for making this change.
